apps/financial-aid/web-osk/src/components/AppLayout/AppLayout.tsx

```diff
@@ -11,18 +12,48 @@ import {
 
 import cn from 'classnames'
 import { AppContext } from '@island.is/financial-aid-web/osk/src/components/AppProvider/AppProvider'
+import { Routes } from '@island.is/financial-aid/shared/lib'
 
 interface Props {
   children: ReactNode
 }
 
 const AppLayout = ({ children }: Props) => {
   const { isAuthenticated, user, loadingUser } = useContext(AppContext)
+  const router = useRouter()
 
   useEffect(() => {
     document.title = 'Fjárhagsaðstoð'
   }, [])
 
+  const getRedirectPath = () => {
+    if (user?.spouse?.hasFiles) {
+      return `${Routes.statusPage(user.currentApplicationId as string)}`
+    }
+    if (user?.spouse?.hasPartnerApplied) {
+      return `${Routes.form.info}`
+    }
+
+    if (user?.currentApplicationId) {
+      return `${Routes.statusPage(user.currentApplicationId as string)}`
+    }
+
+    return `${Routes.application}`
+  }
+
+  if (isAuthenticated && user) {
+    const redirectPath = getRedirectPath()
+
+    if (
+      router.pathname === '/' ||
+      (router.pathname.startsWith(Routes.application) &&
+        !redirectPath.startsWith(Routes.application))
+    ) {
+      router.push(redirectPath)
+      return null
```

Comment:
Is this return needed?

Reply:
It is not necessary but makes the UX a bit better. If we don't have the return then the page we are redirecting from gets displayed for a second, as it takes a bit of time for the redirect to happen.